File: files/2026-03-15/CSEC 15032026.py
import zipfile
import xml.etree.ElementTree as ET
import pandas as pd

path = "2026-03-15/Challenge 107.xlsx"
NS = "http://schemas.openxmlformats.org/spreadsheetml/2006/main"
with zipfile.ZipFile(path) as z:
    with z.open("xl/worksheets/sheet1.xml") as f:
        root = ET.parse(f).getroot()

col_elements = root.findall(f".//{{{NS}}}col")
hidden_cols = [int(col.get("min")) for col in col_elements if col.get("hidden") == "1"]
df = pd.read_excel(path, usecols="A:I", skiprows=1, nrows=4)
test = pd.read_excel(path, usecols="K", skiprows=1, nrows=4)
cols_to_drop = [df.columns[i - 1] for i in hidden_cols]
result = (
    df.drop(columns=cols_to_drop)
      .dropna(axis=1, how="all")
)
result["Total"] = result.select_dtypes(include="number").sum(axis=1)
(result["Total"] == test["Total"].values).all()
# True

File: files/2026-03-15/CSEC 15032026.R
library(xml2)
library(zip)
library(readxl)

path <- "2026-03-15/Challenge 107.xlsx"

unzip(path, exdir = "xlsx")
sheet <- read_xml("xlsx/xl/worksheets/sheet1.xml")
# creates additional files.
cols <- xml_find_all(sheet, ".//d1:col")
hidden <- xml_attr(cols, "hidden")
hidden = data.frame(hidden) %>%
  na.omit() %>%
  tibble::rownames_to_column("col_index")
cols_to_exclude = as.numeric(hidden$col_index)

df = read_excel(path, range = "A2:I6")
test = read_excel(path, range = "K2:K6")

result = df %>%
  select(-cols_to_exclude) %>%
  select(where(~ !all(is.na(.)))) %>%
  rowwise() %>%
  mutate(Total = sum(c_across(where(is.numeric)), na.rm = TRUE))

all.equal(result$Total, test$Total)
# [1] TRUE
